### PR TITLE
Fix constructor to support all Sequelize options

### LINKED ISSUE #119: Logging option not working in 0.5.0.beta.1
Hello,

The logging option it doesn't seem to be working correctly. This is the configuration I'm using to instantiate Sequelize.

```
{ 
      dialect: 'postgres',
      uri: 'postgres://suttna:development@localhost:5432/db_test',
      logging: false,
      modelPaths: [ '/Users/bilby91/Projects/app/bot/src/app/models' ],
      pool: { max: 5, min: 0, idle: 10000 } 
}
```

With that configuration I'm getting log messages every time an interaction with the db is happening.

### PR DESCRIPTION
**Pull request by bilby91**
Fixes #119

### CODE REVIEW

**Comment by bilby91** [comment]
@RobinBuschmann Hey, what do you think ? I think this is the last issue that is blocking us to get this in production! A beta.2 today would be AWESOME!

File: test/utils/sequelize.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ export function createSequelize(useModelsInPath: boolean = true, define: DefineO
 export function createSequelizeValidationOnly(useModelsInPath: boolean = true): Sequelize {
 
   return new Sequelize({
+    dialect: 'sqlite',
```
Comment (RobinBuschmann) [inline test/utils/sequelize.ts]: @bilby91 Whats the purpose of adding the `dialect`option here?

**Comment by bilby91** [comment]
I'm not sure why it was working before. v4 requires a dialect to be explicitly defined.

**Comment by RobinBuschmann** [comment]
Strange, there seem to have happened an issue earlier caused by another pull request I think.

There is a `ISequelizeValidationOnlyConfig` which is missing in:

`export type SequelizeConfig = ISequelizeConfig | ISequelizeUriConfig | ISequelizeDbNameConfig;`

When you add `ISequelizeValidationOnlyConfig` to `SequelizeConfig` it should work without `dialect: 'sqlite'`.

**Comment by bilby91** [comment]
@RobinBuschmann Want me to add it ?

**Comment by RobinBuschmann** [comment]
It depends... does it build without `dialect: 'sqlite',`? Cause this line does not fit the actual api, because it is `validationOnly` without the need to specify database configuration.

**Comment by bilby91** [comment]
@RobinBuschmann Changing this `export type SequelizeConfig = ISequelizeConfig | ISequelizeUriConfig | ISequelizeDbNameConfig | ISequelizeValidationOnlyConfig;` didn't fix the issue. I think the problem is different. The thing is that now the url constructor passes the options always and in v4 you need to always define the dialect. Make sense ?

**Comment by bilby91** [comment]
If we remove the `dialect: 'sqlite'` tests won't pass. And I think it makes sense that they don't pass without that option.

**Comment by RobinBuschmann** [comment]
When `validationOnly: true` is defined, these options (e.g. `dialect`) will be added automatically: https://github.com/RobinBuschmann/sequelize-typescript/blob/master/lib/models/BaseSequelize.ts#L61

**Comment by RobinBuschmann** [comment]
`validationOnly: true` makes it possible to use `sequelize-typescript` without connecting to any database. Thats why it should not be necessary to define database options.

**Comment by RobinBuschmann** [comment]
Here the reason why it compiles with `validationOnly: true` and without `dialect`: https://github.com/RobinBuschmann/sequelize-typescript/blob/master/lib/models/Sequelize.d.ts#L11

**Comment by bilby91** [comment]
@RobinBuschmann Good. Got it. It was not working localy because I though `npm run test` was compiling the code :(.

Now it should be good to go

**Comment by bilby91** [comment]
Thanks for the awesome feedback loop @RobinBuschmann . Can we get a beta.2 🎉 ?

**Comment by RobinBuschmann** [comment]
Thank you for fixing this :) 

Its done `sequelize-typescript@0.5.0-beta.2` 🎉

**Comment by bilby91** [comment]
@RobinBuschmann Thanks!!

**Pull request merged**